torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp

```diff
@@ -127,6 +128,60 @@ RegisterOperators reg(
            };
          },
          aliasAnalysisSpecialCase()),
+     Operator(
+         "prim::AllocateSlab() -> Storage",
+         [](const Node* node) -> Operation {
+           int64_t total_size = node->i(attr::total_size);
+           auto device_type =
+               static_cast<DeviceType>(node->i(attr::device_type));
+           return [total_size, device_type](Stack* stack) {
+             auto allocator = GetAllocator(device_type);
+             auto slab = c10::Storage(
+                 c10::Storage::use_byte_size_t(),
+                 total_size,
+                 allocator,
+                 /*resizable=*/false);
+             push(stack, std::move(slab));
+           };
+         },
+         aliasAnalysisSpecialCase()),
+     Operator(
+         prim::AllocateTensor,
+         [](const Node* node) -> Operation {
+           int64_t size = node->i(attr::size);
```

Comment:
prim::AllocateTensor we probably want to consider to be a write to the input slab... e.g. we dont want to move prim::AllocateTensors around each other, that might mess things up.

We could also pessimistically just call this a side effectful node as below - memory planning is the last optimization we should be doing realistically, so we dont need to worry too much about making things analyzable for subsequent passes

Reply:
>prim::AllocateTensor we probably want to consider to be a write to the input slab... e.g. we dont want to move prim::AllocateTensors around each other, that might mess things up.

I don't think moving them around is a big deal since they carry their offset and size explicitly and as far them writes i don't think that's quite accurate either (the writing doesn't actually happen within this node)

Reply:
The size and offset are dependent on the topology of the network being frozen... the more i'm thinking this we should enforce that this is the final pass, and throw if we see any of these operators in alias analysis

Reply:
> The size and offset are dependent on the topology of the network being frozen... 

yes but not wrt these nodes themselves right? only through the nodes that they feed being dependent on the topology of the graph (i.e. certain input sizes). you could definitely reorder any prefix of the sequence of `AllocateTensor` nodes and it wouldn't make any difference (e.g. you could just bunch up all of the `AllocateTensor` nodes at the beginning of the graph).